core/threaded/reactor_threaded.c

```diff
@@ -1023,7 +1023,9 @@ void _lf_worker_do_work(int worker_number) {
                 current_reaction_to_execute->chain_id,
                 current_reaction_to_execute->deadline);
 
-        bool violation = _lf_worker_handle_violations(
+        // Check for STP violations. Deadline violations checked after acquiring all the
+        //  mutexes.
```

Comment:
Here I remove the check for deadline violation and only check for STP. I believe STP violations are unaffected by the prospect of stalling for an arbitrary period of time between the check and reaction invokation